sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java

```diff
@@ -444,138 +441,98 @@
-            .doOnError(error -> logger.warning("Failed to set read only ConfigurationSetting - {}", setting, error));
-    }
-
-    /**
-     * Clear read-only of the {@link ConfigurationSetting} with a matching {@code key}, and optional {@code label}
-     * combination.
-     *
-     * <p><strong>Code Samples</strong></p>
-     *
-     * <p>Clear read-only of the setting with the key-label "prodDBConnection"-"westUS".</p>
-     *
-     * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.clearReadOnly#string-string}
-     *
-     * @param key The key of configuration setting to clear read-only.
-     * @param label The label of configuration setting to clear read-only, or optionally, null if a setting with
-     * label is desired.
-     * @return The read-only of {@link ConfigurationSetting} that was cleared, or an empty Mono is also returned if
-     * a key collision occurs or the key is an invalid value (which will also throw HttpResponseException described
-     * below).
-     * @throws IllegalArgumentException If {@code key} is {@code null}.
-     * @throws HttpResponseException If {@code key} is an empty string.
-     */
-    @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<ConfigurationSetting> clearReadOnly(String key, String label) {
-        try {
-            return withContext(
-                context -> clearReadOnly(new ConfigurationSetting().setKey(key).setLabel(label), context))
-                .flatMap(response -> Mono.justOrEmpty(response.getValue()));
-        } catch (RuntimeException ex) {
-            return monoError(logger, ex);
-        }
-    }
-
-    /**
-     * Clear read-only of the {@link ConfigurationSetting} with a matching {@link ConfigurationSetting#getKey() key},
-     * and optional {@link ConfigurationSetting#getLabel() label} combination.
-     *
-     * <p><strong>Code Samples</strong></p>
+     * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.setReadOnlyWithResponse#ConfigurationSetting-boolean}
      *
      * <p>Clear read-only of the setting with the key-label "prodDBConnection"-"westUS".</p>
      *
-     * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.clearReadOnlyWithResponse#ConfigurationSetting}
+     * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.setReadOnlyWithResponse#ConfigurationSetting-boolean-clearReadOnly}
      *
-     * @param setting The setting to clear read-only based on its key and optional label combination.
-     * @return A REST response containing the cleared read-only ConfigurationSetting, or {@code null} if didn't exist.
-     * {@code null} is also returned if the {@link ConfigurationSetting#getKey() key} is an invalid value. (which will
-     * also throw HttpResponseException described below).
+     * @param setting The configuration setting to set to read-only or not read-only based on the {@code isReadOnly}.
```

Comment:
Don't need to explain read only here, that will be explained by the parameter itself.